dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/metadata/ConfigurableMetadataServiceExporter.java

```diff
@@ -111,10 +115,17 @@ private ProtocolConfig generateMetadataProtocol() {
                 Protocol protocol = applicationModel.getExtensionLoader(Protocol.class).getExtension(specifiedProtocol);
                 if (protocol != null && protocol.getServers() != null) {
                     Iterator<ProtocolServer> it = protocol.getServers().iterator();
+                    // metadata service may export before normal service export, it.hasNext() will return false.
+                    // so need use specified protocol port.
                     if (it.hasNext()) {
                         String addr = it.next().getAddress();
                         String rawPort = addr.substring(addr.indexOf(":") + 1);
                         protocolConfig.setPort(Integer.parseInt(rawPort));
+                    } else {
+                        Integer protocolPort = getProtocolConfig(specifiedProtocol).getPort();
```

Comment:
Why use `protocol port` instead of specified`port` here?

Reply:
because at here specified port is null, then will use random port to export metadata service, which may not same with specified protocol port, then will cause invoke failed.

Reply:
Please check if `getProtocolConfig(specifiedProtocol)` will return null to cause NPE.

Reply:
if `getProtocolConfig(specifiedProtocol)` return null that means application RPC protocol not set. if so, application will startup failed before export metadataService.

Reply:
It would be better to throw a customized exception if get protocol failed. Others look good to me.

Reply:
I will submit a followup patch right away regarding this small issue.